scipy/stats/_pearsonr.py

```diff
@@ -0,0 +1,329 @@
+import warnings
```

Comment:
For the sake of review, can you move this back into `stats`? After this is merged, I'd be happy to review a trivial PR that simply moves this into its own file.

Reply:
There have been improvements to the docstring that need to be merged here. I will do the surgery.